src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6000,12 +6047,21 @@ void AuraEffect::HandlePeriodicDamageAurasTick(Unit* target, Unit* caster) const
     if (crit)
         damage = caster->SpellCriticalDamageBonus(m_spellInfo, damage, target);
 
+    if (partial && m_period > 0)
+    {
+        int32 partialTickDuration = GetBase()->GetTickRemainingPeriod();
+        float dmgPercent = partialTickDuration / m_period;
```

Comment:
this doesn't work - both partialTickDuration and m_period are integers, producing integer result (always 0 since m_period > partialTickDuration) which is then cast to float
You have to cast one of these values to float before division